### PR TITLE
RDM-5949-added appinsightkey to secrets,migration to AKS changes imple…

### PR DESCRIPTION
**Pull request by akdasari**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[x ] Yes
[ ] No
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # Keep hub.Dockerfile aligned to this file as far as possible
 ARG JAVA_OPTS="-Djava.security.egd=file:/dev/./urandom"
-ARG APP_INSIGHTS_AGENT_VERSION=2.3.1-SNAPSHOT
+ARG APP_INSIGHTS_AGENT_VERSION=2.4.1
```
Comment (smathangi) [inline Dockerfile]: bump to latest 2.5.1-BETA (https://github.com/hmcts/ccd-definition-store-api/pull/458)

File: build.gradle [hunk 2]
```diff
@@ -111,7 +111,8 @@ sourceCompatibility = 1.8
 targetCompatibility = 1.8
 
 ext {
-    reformLogging= '5.0.1'
+    reformLogging= '5.1.0'
```
Comment (smathangi) [inline build.gradle]: bump to latest see (hmcts/ccd-definition-store-api#458)

File: dependency-check-suppressions.xml [hunk 3]
```diff
@@ -140,4 +140,11 @@
         <cve>CVE-2019-12814</cve>
         <cve>CVE-2019-12384</cve>
     </suppress>
+    <suppress until="2019-10-01">
```
Comment (smathangi) [inline dependency-check-suppressions.xml]: remove this as jackson is bumped in build.gradle already

File: charts/ccd-user-profile-api/values.yaml [hunk 4]
```diff
@@ -2,12 +2,17 @@ java:
...
-    IDAM_S2S_URL: http://rpe-service-auth-provider-aat.service.core-compute-aat.internal
-    USER_PROFILE_DB_HOST: "{{ .Release.Name }}-postgresql"
+    IDAM_S2S_URL: http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
+    ENABLE_DB_MIGRATE: false
+    USER_PROFILE_DB_HOST: ccd-user-profile-api-postgres-db-{{ .Values.global.environment }}.postgres.database.azure.com
     USER_PROFILE_DB_PORT: 5432
-    USER_PROFILE_DB_NAME: "{{ .Values.postgresql.postgresqlDatabase}}"
-    USER_PROFILE_DB_USERNAME: "{{ .Values.postgresql.postgresqlUsername}}"
-    USER_PROFILE_DB_PASSWORD: "{{ .Values.postgresql.postgresqlPassword}}"
+    USER_PROFILE_DB_NAME: ccd_user_profile
```
Comment (smathangi) [inline charts/ccd-user-profile-api/values.yaml]: could you double check all variables in main.tf are covered here with correct default values

File: charts/ccd-user-profile-api/values.preview.template.yaml [hunk 5]
```diff
@@ -1,6 +1,19 @@
 java:
-  # Don't modify below here
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
+  aadIdentityName:
+  keyVaults:
+    ccd:
+      resourceGroup: ccd-shared
+      secrets:
+        - user-profile-api-POSTGRES-PASS
```
Comment (smathangi) [inline charts/ccd-user-profile-api/values.preview.template.yaml]: this is not required

File: charts/ccd-user-profile-api/values.yaml (hunk 4 above)
Comment (akdasari) [reply charts/ccd-user-profile-api/values.yaml]: yeah all variables reflect here

File: Dockerfile (hunk 1 above)
Comment (akdasari) [reply Dockerfile]: mmh, done

File: build.gradle (hunk 2 above)
Comment (akdasari) [reply build.gradle]: done

File: Dockerfile [hunk 6]
```diff
@@ -1,13 +1,12 @@
...
-ARG APP_INSIGHTS_AGENT_VERSION=2.3.1-SNAPSHOT
+ARG APP_INSIGHTS_AGENT_VERSION=2.5.1-BETA
 
-FROM hmctspublic.azurecr.io/base/java:openjdk-8-distroless-1.0
+FROM hmctspublic.azurecr.io/base/java:openjdk-8-distroless-1.2
 LABEL maintainer="https://github.com/hmcts/ccd-user-profile-api"
 
 COPY build/libs/user-profile.jar  /opt/app/
 COPY lib/AI-Agent.xml /opt/app
-COPY lib/applicationinsights-agent-2.3.1-SNAPSHOT.jar lib/AI-Agent.xml /opt/app/
```
Comment (mario-paniccia) [inline Dockerfile]: you added applicationinsights-agent-2.5.1-BETA.jar. Is this required? Isn't this automatically downloaded by the base image?

File: build.gradle [hunk 7]
```diff
@@ -111,7 +111,8 @@ sourceCompatibility = 1.8
 targetCompatibility = 1.8
 
 ext {
-    reformLogging= '5.0.1'
+    reformLogging= '5.1.1-BETA'
+    appInsightsVersion = '2.4.1'
```
Comment (mario-paniccia) [inline build.gradle]: does this needs to be updated to 2.5.1-BETA?

File: build.gradle [hunk 8]
```diff
@@ -158,11 +159,17 @@ dependencies {
     compile 'net.jcip:jcip-annotations:1.0'
     compile 'com.github.spotbugs:spotbugs-annotations:3.1.12'
 
+    // redefining with version 2.4.1 as applicationinsights-agent-2.5.0 has an open issue
```
Comment (mario-paniccia) [inline build.gradle]: does 2.5.1-BETA has the issue as well? if not, can we use 2.5.1-BETA here for consistency with the buildfile?

File: charts/ccd-user-profile-api/values.aat.template.yaml [hunk 9]
```diff
@@ -1,12 +1,5 @@
 java:
+  replicas: 2
+  useInterpodAntiAffinity: true
```
Comment (mario-paniccia) [inline charts/ccd-user-profile-api/values.aat.template.yaml]: we sure we want to use interpodAffinity?

File: build.gradle (hunk 8 above)
Comment (smathangi) [reply build.gradle]: no need to redefine them here as java-logging will transitively get 2.5.1-BETA

File: Dockerfile (hunk 6 above)
Comment (mario-paniccia) [reply Dockerfile]: confirmed with Tim. That is correct, but we still need it there because it's needed in the ASE envs.

Comment (akdasari) [reply Dockerfile]: Still need this for  ASE, confirmed from Praveen

**Pull request merged**